doc/ocpf-match-action.md

```diff
@@ -0,0 +1,385 @@
+
+This proposal defines an order of operations which allow match action rules to coexist with OC’s traditional ACL, QoS classifiers, policing, queuing, TCAM-based match and forward rules, and LPM-based match and forward rules.
+
+# 3. OC Model Design
+
+This design expands the policy-forwarding tree with more rules and actions and a new policy type, MATCH_ACTION_POLICY.
+
+If we apply an OCPF policy with actions to accept/drop and set forwarding-group to an interface, it would be wise to not define any ACL or QoS policy to that same interface to prevent ambiguity.  To ensure there is no ambiguity regarding mixing of ACL, QOS and OCPF MATCH-ACTION on a single interface, the   /policy-forwarding/policies/policy/config/type  MATCH_ACTION_POLICY is introduced.  When set, the implementation should reject configuration of other OCPF types, ACL or QOS classifier on a given interface.
+
+## 3.1. policy-forwarding tree additions
+
+The following tree summarizes the leaves to be added to the model.  This fake
+rendering will be replaced with a real rendering of the yang tree.
+
+```diff
+  +--rw network-instances
+     +--rw network-instance*[name]
+        +--rw policy-forwarding
+        |  +--rw policies
+        |  |  +--rw policy* [policy-id]
+        |  |     +--rw config
++                     +--rw type?    # Add enum MATCH_ACTION
+        |  |     +--rw rules
+        |  |        +--rw rule* [sequence-id]
++                      +--rw type?              -> oc-pf:rule-type # identity ref, similar to /acl/type
++                      +--rw name?              string  # user defined name for a rule
+                       +--rw l2?
++                      +--rw mpls-traffic-class?     oc-mplst:mpls-tc
+                       +--rw ipv4
+                          +--rw source-address?
+                          +--rw destination-address?
+                          +--rw source-address-prefix-set?
+                          +--rw destination-address-prefix-set?
++                         +--rw dscp?                oc-inet:dscp
+                          +--rw dscp-set?
+                          +--rw hop-limit?
+                          +--rw fragment-offset?
++                         +--rw options?             oc-inet:options
++                         +--rw precedence?          oc-inet:precedence
+                          +--rw protocol?
+                       +--rw ipv6
+                          +--rw hop-limit?
+                          +--rw destination-address?
+                          +--rw destination-address-prefix-set?  # exclude use case public#1314
+                          +--rw fragment-offset?
+                          +--rw protocol?
+                          +--rw source-address?
+                          +--rw source-address-prefix-set?
+                       +--rw action
+                          +--rw config
++                            +--rw count?              string  #user defined counter
```

Comment:
Should this not be a `leafref` to one of the defined counters that are referenced below?